src/api/services/messages-service.ts

```diff
@@ -0,0 +1,85 @@
+import {ApiRoot} from '../persistence/ApiRoot';
+import {CoreWebService} from './core-web-service';
+import {Injectable} from '@angular/core';
+import {LoginService} from './login-service';
+import {Observable} from 'rxjs/Observable';
+import {RequestMethod, Http} from '@angular/http';
+import {Subject} from 'rxjs/Subject';
+import _ from 'lodash';
+
+@Injectable()
+export class MessageService extends CoreWebService {
+    private _messageMap$: Subject<any> = new Subject<any>();
+    private doMessageLoad;
+    private i18nUrl: string;
+    private lang: string;
+    private messageKeys: Array;
+    private messagesLoaded: any;
+
+    constructor(apiRoot: ApiRoot, http: Http, loginService: LoginService) {
+        super(apiRoot, http);
+        this.i18nUrl = 'v1/languages/i18n';
```

Comment:
the url could be encapsulated in an interface/factory, if the version changed we would like to avoid to change on all url over the app the v1 or even worst change in some of them, and some of them not.

Eventually we have to think in the possibility to support components by version